src/lazy/never.rs

```diff
@@ -31,3 +37,128 @@ impl<'top, D: LazyDecoder> From<Never> for MacroExpr<'top, D> {
         unreachable!("macro in Ion 1.0 (method: into)")
     }
 }
+
+impl AnnotatableValueWriter for Never {
```

Comment:
🗺️ Implementing all of these traits for `Never` allows it to be used to satisfy macro-related associated types in Ion 1.0, statically preventing those types/methods from being used.